File: app/api/token/token.js
'use strict';

async function getToken() {
    try {
        // Use dynamic import with await
        const { default: fetch } = await import('node-fetch');

        const API_KEY_SECRET = 'cleverattisfu_default_secret';
        const cleveratti_URL = 'https://sfu.cleveratti.com/api/v1/token';
        //const cleveratti_URL = 'http://localhost:3010/api/v1/token';

        const response = await fetch(cleveratti_URL, {
            method: 'POST',
            headers: {
                authorization: API_KEY_SECRET,
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                username: 'username',
                password: 'password',
                presenter: true,
                expire: '1h',
            }),
        });
        const data = await response.json();
        if (data.error) {
            console.log('Error:', data.error);
        } else {
            console.log('token:', data.token);
        }
    } catch (error) {
        console.error('Error fetching data:', error);
    }
}

getToken();

File: app/api/token/token.py
# pip3 install requests
import requests
import json

API_KEY_SECRET = "cleverattisfu_default_secret"
cleveratti_URL = "https://sfu.cleveratti.com/api/v1/token"
#cleveratti_URL = "http://localhost:3010/api/v1/token"

headers = {
    "authorization": API_KEY_SECRET,
    "Content-Type": "application/json",
}

data = {
    "username": "username",
    "password": "password",
    "presenter": "true",
    "expire": "1h"
}

response = requests.post(
    cleveratti_URL, 
    headers=headers, 
    json=data
)

print("Status code:", response.status_code)
data = json.loads(response.text)
print("token:", data["token"])
